src/gui/touchscreengui.cpp

```diff
@@ -860,74 +769,36 @@ void TouchScreenGUI::translateEvent(const SEvent &event)
 				if (distance <= m_touchscreen_threshold) {
 					m_joystick_speed = 0.0f;
 				} else {
-					m_joystick_speed = distance / button_size;
+					m_joystick_speed = distance / m_button_size;
 					if (m_joystick_speed > 1.0f)
 						m_joystick_speed = 1.0f;
 				}
 
 				m_joystick_status_aux1 = distance > (half_button_size * 3);
 
-				if (distance > button_size) {
+				if (distance > m_button_size) {
 					// move joystick "button"
-					v2s32 new_offset = dir * button_size / distance - half_button_size;
+					v2s32 new_offset = dir * m_button_size / distance - half_button_size;
 					if (m_fixed_joystick)
-						m_joystick_btn_center->gui_button->setRelativePosition(
+						m_joystick_btn_center->setRelativePosition(
 								fixed_joystick_center + new_offset);
 					else
-						m_joystick_btn_center->gui_button->setRelativePosition(
+						m_joystick_btn_center->setRelativePosition(
 								free_joystick_center + new_offset);
 				} else {
-					m_joystick_btn_center->gui_button->setRelativePosition(
+					m_joystick_btn_center->setRelativePosition(
 							touch_pos - half_button_size);
 				}
 			}
 		}
-
-		if (!m_has_move_id && !m_has_joystick_id)
-			handleChangedButton(event);
-	}
-}
-
-void TouchScreenGUI::handleChangedButton(const SEvent &event)
```

Comment:
I decided against reimplementing `handleChangedButton` for these reasons:

1. It was likely introduced for use with the D-pad, which doesn't exist anymore.

2. The current behavior is hardly useful and usually not noticeable because it's only applied when `!m_has_move_id && !m_has_joystick_id`.

3. If we add on-screen dig & place buttons (see #13229), we should allow moving the camera using the finger that holds the dig/place button, without accidentally switching to another button.\*

4. I'm lazy.

\* This wouldn't happen with the current behavior, but it would happen if we fixed (2).